### PR TITLE
feat: add function calling recipe

### LINKED ISSUE #562: Add a recipe that works with function calling models
In podman desktop ai lab we are planning to add function calling models support -> https://github.com/containers/podman-desktop-extension-ai-lab/issues/1116

We would need a recipe + a list of recommended models to work with

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #562

https://github.com/user-attachments/assets/6461b03b-7166-41be-bb0d-d2d4a149e41d

### CODE REVIEW

**Comment by jeffmaury** [comment]
@slemeur @firewall @axel7083 FYI

**Review by axel7083 — COMMENTED** [review]
Really like to see the progress, the recipe works fine by combining https://github.com/containers/podman-desktop-extension-ai-lab/pull/1773 and https://github.com/containers/podman-desktop-extension-ai-lab/pull/1779.

I think we can iterate to improve the the recipe in following PR

File: recipes/natural_language_processing/function_calling/app/app.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+from langchain_core.pydantic_v1 import BaseModel, Field
+from langchain_core.output_parsers import PydanticToolsParser
+from langchain.globals import set_debug, set_verbose
+import streamlit as st
+import requests
+import time
+import json
+import os
+
+from urllib3 import request
```
Comment (axel7083) [inline recipes/natural_language_processing/function_calling/app/app.py]:
```suggestion
from urllib3 import request
```

unused

File: recipes/natural_language_processing/function_calling/app/app.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
+from langchain_openai import ChatOpenAI
+from langchain.chains import LLMChain
+from langchain_core.prompts import ChatPromptTemplate, MessagesPlaceholder
+from langchain_core.pydantic_v1 import BaseModel, Field
+from langchain_core.output_parsers import PydanticToolsParser
+from langchain.globals import set_debug, set_verbose
```
Comment (axel7083) [inline recipes/natural_language_processing/function_calling/app/app.py]:
```suggestion

```

unused

File: recipes/natural_language_processing/function_calling/app/app.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
+from langchain_openai import ChatOpenAI
+from langchain.chains import LLMChain
+from langchain_core.prompts import ChatPromptTemplate, MessagesPlaceholder
```
Comment (axel7083) [inline recipes/natural_language_processing/function_calling/app/app.py]:
```suggestion
from langchain_core.prompts import ChatPromptTemplate
```

unused

File: recipes/natural_language_processing/function_calling/app/app.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
+from langchain_openai import ChatOpenAI
+from langchain.chains import LLMChain
```
Comment (axel7083) [inline recipes/natural_language_processing/function_calling/app/app.py]:
```suggestion
```

unused

**Comment by rhatdan** [comment]
@MichaelClifford @sallyom PTAL

**Comment by rhatdan** [comment]
@slemeur @Firewall @axel7083 @MichaelClifford @sallyom PTAL

**Review by axel7083 — APPROVED** [review]
LGTM

**Pull request merged**